thunder/core/symbol.py

```diff
@@ -527,7 +527,9 @@ def __eq__(self, other):
         return (self.sym, self._var_args, self._var_output) == (other.sym, other._var_args, other._var_output)
 
     def rhs(self):
-        return BoundSymbolRHS(self)
+        return BoundSymbolRHS(
+            self.sym, tuple(tree_flatten(self._var_args)[0]), tuple(tree_flatten(self._var_kwargs)[0])
```

Comment:
`tree_flatten(self._var_kwargs)` removes information about the keys of the dictionary. I don't think it's correct to consider `{correction: 1}` and `{dim: 1}` as equal.

Reply:
Woops, I missed that! Thanks for pointing that out 🙏